apps/condo/domains/billing/schema/fields/AmountDistribution.js

```diff
@@ -0,0 +1,223 @@
+const Ajv = require('ajv')
+const Big = require('big.js')
+const { omit, get, set } = require('lodash')
+
+const { GQLError, GQLErrorCode: { BAD_USER_INPUT } } = require('@open-condo/keystone/errors')
+const { find } = require('@open-condo/keystone/schema')
+
+const {
+    AMOUNT_DISTRIBUTION_FIELD_NAME,
+    AMOUNT_DISTRIBUTION_INPUT_NAME,
+    AMOUNT_DISTRIBUTION_SCHEMA_FIELD,
+    AMOUNT_DISTRIBUTION_SCHEMA_INPUT,
+} = require('@condo/domains/acquiring/constants/gql')
+const {
+    RECIPIENT_FIELD_NAME,
+    RECIPIENT_INPUT_NAME,
+    RECIPIENT_FIELDS_DEFINITION,
+    RecipientSchema,
+} = require('@condo/domains/acquiring/schema/fields/Recipient')
+const {
+    hasOverpaymentReceivers,
+    hasSingleVorItem,
+    hasFeePayers,
+} = require('@condo/domains/acquiring/utils/billingCentrifuge')
+const {
+    WRONG_AMOUNT_DISTRIBUTION_ERROR_TYPE,
+    AMOUNT_DISTRIBUTION_FIELD_NO_APPROVED_BANK_ACCOUNT_ERROR_TYPE,
+    AMOUNT_DISTRIBUTION_FIELD_SUMS_NOT_MATCH_ERROR_TYPE,
+    AMOUNT_DISTRIBUTION_FIELD_NO_VOR_IN_GROUP_ERROR_TYPE,
+    AMOUNT_DISTRIBUTION_FIELD_NO_FEE_PAYER_ERROR_TYPE,
+    AMOUNT_DISTRIBUTION_FIELD_NO_OVERPAYMENT_RECEIVER_ERROR_TYPE,
+} = require('@condo/domains/billing/constants/errors')
+const { render, getGQLErrorValidator } = require('@condo/domains/common/schema/json.utils')
+const { SERVICE } = require('@condo/domains/user/constants/common')
+
+const DEFAULT_TO_PAY_FIELD_NAME = 'toPay'
+
+const ERRORS = {
+    NO_APPROVED_BANK_ACCOUNT: (notApprovedRecipients) => ({
```

Comment:
TODO: we need to talk about this syntax...

Reply:
Made objects, but this leads to bad DX because the developer needs to remember variable names, which must be the objects' keys. Also, parts of the error object are placed in different places of code.